### PR TITLE
Add more info to `OverlayState.insert` error messages

### PR DESCRIPTION
**Pull request by LongCatIsLooong**
I was debugging an Overlay issue and felt I could have identified the problem faster if the existing assertions provided more information about the current state of the OverlayEntry and Overlay.

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/widgets/overlay.dart [hunk 1]
```diff
@@ -543,6 +543,55 @@ class OverlayState extends State<Overlay> with TickerProviderStateMixin {
     return _entries.length;
   }
 
+  bool _debugCanInsertEntry(OverlayEntry entry) {
+    final List<DiagnosticsNode> operandsInformation = <DiagnosticsNode>[
```
Comment (chunhtai) [inline packages/flutter/lib/src/widgets/overlay.dart:547]: probably wrap the whose thing in assert as well, `_debug` prefix should be noop if call directly in debug mode. Not sure if there is a style guide for this, but that is what other _debug method behavior else where in the framework.

File: packages/flutter/lib/src/widgets/overlay.dart [hunk 2]
```diff
@@ -222,7 +222,7 @@ class OverlayEntry implements Listenable {
   }
 
   @override
-  String toString() => '${describeIdentity(this)}(opaque: $opaque; maintainState: $maintainState)';
+  String toString() => '${describeIdentity(this)}(opaque: $opaque; maintainState: $maintainState)${_disposedByOwner ? "(DISPOSED)" : ""}';
```
Comment (chunhtai) [inline packages/flutter/lib/src/widgets/overlay.dart:225]: This may be a bit confused. I think it is possible that it is waiting to be disposed?

**Review by chunhtai — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/widgets/overlay.dart (hunk 1 above)
Comment (LongCatIsLooong) [reply packages/flutter/lib/src/widgets/overlay.dart:547]: Sounds like https://github.com/dart-lang/sdk/issues/28113 ? I think I can try adding an annotation (e.g., `@debugAssert`) and prevent it from being called outside of an `assert` in analyze.dart?

File: packages/flutter/lib/src/widgets/overlay.dart (hunk 2 above)
Comment (LongCatIsLooong) [reply packages/flutter/lib/src/widgets/overlay.dart:225]: `_disposedByOwner` invalidates the `OverlayEntry` immediately. I think the only operation allowed on a disposed OverlayEntry is `removeListener` (same as `ChangeNotifier`).

File: packages/flutter/lib/src/widgets/overlay.dart (hunk 1 above)
Comment (chunhtai) [reply packages/flutter/lib/src/widgets/overlay.dart:547]: this is new to me, if it works, i fine with it

**Comment by goderbauer** [comment]
(Triage): @LongCatIsLooong is this ready to be merged?

**Comment by LongCatIsLooong** [comment]
I'd like to merge the `@debugAssert` PR first so I can mark the new methods as debug-only.

**Comment by goderbauer** [comment]
@LongCatIsLooong Should we just submit this for now? Since the methods are all private, I am not too worried about the missing annotation.

**Pull request merged**